Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java

```diff
@@ -101,4 +107,19 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
                     MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
         }
     }
+
+    private void refreshBidderToken(final Context context) {
+        if (isComputingToken.compareAndSet(false, true)) {
+            new Thread(new Runnable() {
```

Comment:
Can we do it on executor? if getBiddingToken is called too frequently there will be big changes to get OOM...

Reply:
Each thread will wait for the previous one due to blocking logic in the repository anyway.

Reply:
> Can we do it on executor? if getBiddingToken is called too frequently there will be big changes to get OOM...

There's only one thread running, think there's no OOM issue due to isComputingToken check.

Reply:
> Each thread will wait for the previous one due to blocking logic in the repository anyway.

Yes, in AND-3191 ticket we think timeout is one of the main reasons, so I updated the sdk logic. But in my opinion, if we query bid token from DB spend more than 4 seconds, there's still no result even remove the timeout limitation.

Reply:
It's just a public API and we are not in control of its usage.
Each thread consumes a lot of memory and is not GCted immediately.